Fixes for 0.1.6

thepetk opened the pull request:
### What does this PR do?:

Fixes the change made to pvc so it is only created if vllm is selected.

Updates the logic of model-server sample test so it checks a specific endpoint once the model server pod is up and running.

### Which issue(s) this PR fixes:

N/A

### PR acceptance criteria:

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened and linked to this PR, if they are not in the PR scope due to various constraints.

- [ ] Tested and Verified

<!-- _I have verified that the changes were tested manually_ -->

- [ ] Documentation (READMEs, Product Docs, Blogs, Education Modules, etc.)

 <!-- _This includes READMEs, Product Docs, Blogs, Education Modules, etc._ -->

### How to test changes / Special notes to the reviewer:

The linting check will fail because the chart version is already 0.1.6 though I think we should keep it as is.

I've test the current updates with the charts verifier tool and both tests are succesfull.

thepetk committed:
Fix pvc conditional

thepetk committed:
Update model server tests

Jdubrick commented on charts/ai-software-templates/model-server/templates/tests/test-connection-server.yaml:
```diff
@@ -8,8 +8,16 @@ metadata:
-      command: ["wget"]
-      args: ["{{ .Release.Name }}-model-server:{{ .Values.model.modelServicePort }}/"]
+    - name: curl
+      image: curlimages/curl:latest
```
Nit, up to you if you want to implement it:

Should we use the `quay.io` hosted version of curl? For clarity of where it is coming from? 
https://quay.io/repository/curl/curl

thepetk replied:
Yeah I agree let's be consistent with the image used for chatbot.

Jdubrick commented on charts/ai-software-templates/model-server/templates/tests/test-connection-server.yaml:
```diff
@@ -8,8 +8,16 @@ metadata:
+        - >
+          for i in {1..5}; do
+            curl -sSf http://{{ .Release.Name }}-model-server:{{ .Values.model.modelServicePort }}/v1/models && exit 0;
+            echo 'Retrying in 5s...';
```
If this is from a timeout I am in favour of adding an explanation of why we are retrying

michael-valdron replied:
+1 also think we should include the counter which attempt out of how many attempts we will retry. E.g.

```
[2 out of 5] Retrying in 5s...
```

thepetk replied:
Tbh this is not needed in general. Now that the pvc is handled correctly it won't need any retries so I've removed it.

Jdubrick reviewed (commented):
Small comment but looks good

michael-valdron reviewed (commented):
@thepetk Lgtm once Jordan's comments and my additional comment about the timeout are addressed.

thepetk committed:
Update test-connection-server test

thepetk commented:
Addressed the feedback here: [cfe63ed](https://github.com/redhat-ai-dev/ai-lab-helm-charts/pull/37/commits/cfe63ed0363415793a90f00646d1412e69ca326e)

FYI to give a bit of more context the 2 tests currently included in the `chatbot-ai-sample` and `models-server-sample` charts are quite simplistic so the [chart-verifier](https://github.com/redhat-certification/chart-verifier/blob/8ceef0e23aa4660658cae38b92c3015b0e8eb463/docs/helm-chart-checks.md#checks-1) checks are ok.

I don't think this is sufficient tho so one of the reasons I've opened RHDHPAI-615 is to enhance their logic and add them to our test automation.

cc @michael-valdron @Jdubrick (also thanks for the review <3)

Jdubrick reviewed (approved):
lgtm
